src/plugins/explore/public/application/utils/state_management/actions/query_actions.ts

```diff
@@ -63,6 +74,150 @@ export const defaultPrepareQueryString = (query: Query): string => {
+ */
+const createHistogramConfigWithInterval = (
+  dataView: DataView,
+  interval: string | undefined,
+  services: ExploreServices,
+  getState: () => RootState
+): HistogramConfig | null => {
+  if (!dataView.timeFieldName || !interval) {
+    return null;
+  }
+
+  const state = getState();
+  const effectiveInterval = interval || state.legacy?.interval || 'auto';
+
+  const histogramConfigs = createHistogramConfigs(dataView, effectiveInterval, services.data);
+  const aggs = histogramConfigs?.toDsl();
+
+  if (!aggs) {
+    return null;
+  }
+
+  const { fromDate, toDate } = formatTimePickerDate(
+    services.data.query.timefilter.timefilter.getTime(),
+    'YYYY-MM-DD HH:mm:ss.SSS'
+  );
+
+  return {
+    histogramConfigs,
+    aggs,
+    effectiveInterval,
+    fromDate,
+    toDate,
+    timeFieldName: dataView.timeFieldName,
+  };
+};
+
+export const buildQuery = (
+  query: string,
+  histogramConfig: HistogramConfig,
+  services: ExploreServices
+): string => {
+  const { aggs, fromDate, toDate, timeFieldName } = histogramConfig;
+
+  if (!aggs || !timeFieldName) {
+    return query;
+  }
+
+  const finalInterval =
+    aggs[2].date_histogram.fixed_interval ??
+    aggs[2].date_histogram.calendar_interval ??
+    services.data.search.aggs.calculateAutoTimeExpression({
+      from: fromDate,
+      to: toDate,
+      mode: 'absolute',
+    });
+
+  return `${query} | stats count() by span(${timeFieldName}, ${finalInterval})`;
+};
+
+export const processRawResultsForHistogram = (
+  queryString: string,
+  rawResults: ISearchResult,
+  histogramConfig: HistogramConfig
+) => {
+  const { aggs } = histogramConfig;
+
+  if (!aggs) {
+    return rawResults;
+  }
+
+  const aggsConfig: any = {};
+
+  Object.entries(aggs as Record<number, any>).forEach(([key, value]) => {
+    const aggTypeKeys = Object.keys(value);
+    if (aggTypeKeys.length === 0) {
+      return aggsConfig;
+    }
+    const aggTypeKey = aggTypeKeys[0];
+    if (aggTypeKey === 'date_histogram') {
+      aggsConfig[aggTypeKey] = {
+        ...value[aggTypeKey],
+      };
+      aggsConfig.qs = { [key]: queryString };
+    }
+  });
+
+  const responseAggs: any = {};
+
+  for (const [key, _aggQueryString] of Object.entries(aggsConfig.qs)) {
+    responseAggs[key] = rawResults?.hits.hits.map((hit) => {
+      if (rawResults?.fieldSchema && rawResults.fieldSchema.length >= 2) {
+        const valueField = rawResults.fieldSchema[0].name!;
+        const keyField = rawResults.fieldSchema[1].name!;
+        return {
+          key: hit._source[keyField],
+          value: hit._source[valueField],
+        };
+      }
+      const sourceValues = Object.values(hit._source);
+      return {
+        key: sourceValues[1],
+        value: sourceValues[0],
+      };
+    });
+  }
+
+  const tempResult: ISearchResult = { ...rawResults, aggregations: {} };
+
+  Object.entries(responseAggs).forEach(([id, value]) => {
```

Comment:
Instead of creating a `responseAggs` in line 165 and then deconstructing it here to create the `tempResults` with timestamped buckets, I think we should just create `tempResults` with the timestamped buckets from the get-go. 

Having looked at the code for how the `ppl_search_strategy` deconstructs the result into k/v pairs and then the dataframe later does the below bucketing, I understand why it was put in this way, but since this function is very directly creating the histogram, let's just make it as simple as possible.